skimage/_build.py

```diff
@@ -44,48 +43,13 @@ def cython(pyx_files, working_path=''):
         print("Cython >= %s not found; falling back to pre-built %s" \
               % (CYTHON_VERSION, " ".join(c_files)))
     else:
-        for pyxfile in [os.path.join(working_path, f) for f in pyx_files]:
-
-            # if the .pyx file stayed the same, we don't need to recompile
-            if not _changed(pyxfile):
-                continue
-
+        pyx_files = [os.path.join(working_path, f) for f in pyx_files]
+        for i, pyxfile in enumerate(pyx_files):
             if pyxfile.endswith('.pyx.in'):
                 process_tempita_pyx(pyxfile)
-                pyxfile = pyxfile.replace('.pyx.in', '.pyx')
-
-            cythonize(pyxfile)
-
-
-def _md5sum(f):
-    m = hashlib.new('md5')
-    while True:
-        # Hash one 8096 byte block at a time
-        d = f.read(8096)
-        if not d:
-            break
-        m.update(d)
-    return m.hexdigest()
-
-
-def _changed(filename):
-    """Compare the hash of a Cython file to the cached hash value on disk.
-
-    """
-    filename_cache = filename + '.md5'
-
-    try:
-        md5_cached = open(filename_cache, 'rb').read()
-    except IOError:
-        md5_cached = '0'
-
-    with open(filename, 'rb') as f:
-        md5_new = _md5sum(f)
-
-        with open(filename_cache, 'wb') as cf:
-            cf.write(md5_new.encode('utf-8'))
+                pyx_files[i] = pyxfile.replace('.pyx.in', '.pyx')
 
-    return md5_cached != md5_new.encode('utf-8')
+        cythonize(pyx_files, nthreads=8)
```

Comment:
I'm not sure about hardcoding the number of threads. I mean this looks like a reasonable default, but I'm just not sure :).

Reply:
Let me experiment with this number tonight. Cython might use ncores*2 automatically.

Reply:
@hmaarrfk Doesn't look so - https://github.com/cython/cython/blob/a0bbb940c847dfe92cac446c8784c34c28c92836/Cython/Build/Dependencies.py#L923-L925

Reply:
I added a note for it.